pkg/observer.go

```diff
@@ -0,0 +1,69 @@
+package ocr2keepers
+
+import "context"
+
+type Tick interface {
+	// GetUpkeeps provides an array of upkeeps scoped to the individual tick
+	GetUpkeeps(ctx context.Context) ([]UpkeepPayload, error)
+}
+
+// Preprocessor is the general interface for middleware used to filter, add, or modify upkeep
+// payloads before checking their eligibility status
+type Preprocessor interface {
+	// PreProcess takes a slice of payloads and returns a new slice
+	PreProcess(context.Context, []UpkeepPayload) ([]UpkeepPayload, error)
+}
+
+// Postprocessor is the general interface for a processing function after checking eligibility status
+type Postprocessor interface {
+	// PostProcess takes a slice of results where eligibility status is known
+	PostProcess(context.Context, []CheckResult) error
+}
+
+// Runner2 is the interface for an object that should determine eligibility state
+type Runner2 interface {
```

Comment:
what is runner1?

Reply:
`Runner2` is a type that was introduced as part of the OCR3 fan out work and can be ignored as part of the refactor review; unfortunately the PR for this change was merged into develop before I could get this PR open

Reply:
my question was more about why this is called runner2 :sweat_smile: ant not just runner

Reply:
Haha, okay so there's a Runner component that we use in v2.0. When the OCR3-related PR was merged into develop, the chainlink build was failing because there was an interface change in the Runner component. For this reason, Runner2 was created to sidestep the chainlink build issue. So this is a temporary name for what will eventually be Runner. The problem here is that develop was used as the refactor base branch, but that has now extended into being used as the OCR3 branch.

Reply:
This can be moved to a temporary v2 package just to make the separation clear. We just have to move 2 files